FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// Defines the required continue/state URL with optional Android and iOS settings. Used when
+    /// invoking the email action link generation APIs in <see cref="FirebaseAuth"/>.
+    /// </summary>
+    public sealed class ActionCodeSettings
+    {
+        /// <summary>
+        /// Gets or sets the continue/state URL. This property has different meanings in different
+        /// contexts.
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// When the link is handled in the web action widgets, this is the deep link in the
+        /// <c>continueUrl</c> query parameter.
+        /// </description>
+        /// </item>
+        /// <item>
+        /// <description>
+        /// When the link is handled in the app directly, this is the <c>continueUrl</c> query
+        /// parameter in the deep link of the Dynamic Link.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// This property is required.
+        /// </summary>
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to open the link via a mobile app or a browser.
+        /// The default is false. When set to true, the action code link is sent as a Universal
+        /// Link or an Android App Link and is opened by the app if installed. In the false case,
```

Comment:
Is any of this relevant when the app isn't installed?  I wonder if we need this phrase.

Reply:
I think those are orthogonal issues. If the app isn't installed, you will still get a dynamic link that will offer to install the app.